### PR TITLE
bugFix and exclude empty evidence from saving

### PR DESCRIPTION
**Pull request by JiaoJiao123**

### CODE REVIEW

File: web/yo/app/scripts/controllers/gene.js [hunk 1]
```diff
@@ -2211,12 +2215,15 @@ angular.module('oncokbApp')
                     data.knownEffect = 'Resistant';
                     break;
                 case 'CLINICAL_TRIAL':
-                    dataUUID = tumor.trials_uuid.getText();
+                    dataUUID = '';
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: You already set it at line 2980

File: web/yo/app/scripts/controllers/gene.js [hunk 2]
```diff
@@ -2106,10 +2106,13 @@ angular.module('oncokbApp')
                     }
                 }
             }
-
+            function getValidUUID(uuid, content) {
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: The function name doesn't explain the purpose of this function

File: web/yo/app/scripts/controllers/gene.js [hunk 3]
```diff
@@ -2614,11 +2621,12 @@ angular.module('oncokbApp')
                     typeArr = [ti.name.getText()];
                     dataArr = ti.treatments.asArray();
                     tempType = 'treatment';
+                    unObsoletedEvidences(typeArr, mutation, tumor, ti, null, evidences);
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: This name does not explain the purpose of it

File: web/yo/app/scripts/controllers/gene.js (hunk 1 above)
Comment (JiaoJiao123) [inline web/yo/app/scripts/controllers/gene.js]: maybe you mean any other line? since 2980 is an empty line

**Pull request merged**